clearwater-infrastructure/usr/share/clearwater/bin/check-uptime

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+
+# @file check-uptime
+#
+# Project Clearwater - IMS in the Cloud
+# Copyright (C) 2013  Metaswitch Networks Ltd
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+# Checks that a process has been running for at least REQUIRED_UPTIME
+# seconds. Returns 0 if it has. Returns a non-zero value if it has not.
+REQUIRED_UPTIME=30
+
+# Bail out and return non-zero if anything goes wrong.
+set -e
+
+# The command-line argument is the pidfile for the process.
+[ $# = 1 ] || { echo "Usage: check-uptime <pidfile>" >&2 ; exit 2 ; }
+pidfile=$1
+
+# It's expected that there might not be a pidfile.
+pid=$( cat $pidfile 2>/dev/null)
+
+value=$( ps -p $pid -o etimes= 2>/dev/null) || { echo "No process matching value from pidfile: $pid" >&2 ; exit 1 ; }
+if [ "$value" -gt "$REQUIRED_UPTIME" ]; then
```

Comment:
how accurate will '$value' be? 
If it only resolves to the nearest second, should we be making the comparison -ge rather than -gt, so that we aren't requiting an uptime of 31 seconds?

To be honest, i doubt the extra second matters, but it makes it feel cleaner if we are checking uptime of exactly 30 seconds.